src/app/shared/quest.service.ts

```diff
@@ -23,11 +24,31 @@ export class QuestService {
   }
 
   addQuest(quest: IQuest): Observable<IQuest> {
-    return this.http
-      .post<IQuest>(BASIC_URL, quest)
-      .pipe(tap(newQuest => {
+    return this.http.post<IQuest>(BASIC_URL, quest).pipe(
+      tap(newQuest => {
         const updatedQuests = [...this.quests.getValue(), newQuest];
         this.quests.next(updatedQuests);
-      }));
+      })
+    );
+  }
+
+  acceptQuest(id: number): Observable<IQuest> {
+    const acception = {
+      isAccepted: true
+    };
+
+    return this.http.patch<IQuest>(`${BASIC_URL}/${id}`, acception).pipe(
```

Comment:
You don't need the whole quest for the patch request. You can send only changed value